### ISSUE TITLE
Simplify tokenizer

### ISSUE THREAD

**Issue by Kronuz** [bastikr/boolean.py#34]
I like this tokenizer better than using `tokenize`.

 I thought standard tokenize module would be optimized in C, but it's fully written in Python and since it has so much more than we actually need, I'm sure it's not as efficient. Furthermore, I'd go as far as yielding the token, tokstr and the position, it being either an offset or a (row, col) pair, which for most cases would be enough:

```python
def tokenizer(expr, symbol_class=boolean.Symbol):
    if not isinstance(expr, basestring):
        raise TypeError("expr must be string but it is %s." % type(expr))

    TOKENS = {
        '*': boolean.TOKEN_AND,
        '&': boolean.TOKEN_AND,
        'and': boolean.TOKEN_AND,
        '|': boolean.TOKEN_OR,
        '+': boolean.TOKEN_OR,
        'or': boolean.TOKEN_OR,
        '~': boolean.TOKEN_NOT,
        '!': boolean.TOKEN_NOT,
        'not': boolean.TOKEN_NOT,
        '(': boolean.TOKEN_LPAR,
        '[': boolean.TOKEN_LPAR,
        ']': boolean.TOKEN_RPAR,
        ')': boolean.TOKEN_RPAR,
        'true': boolean.TRUE,
        '1': boolean.TRUE,
        'false': boolean.FALSE,
        '0': boolean.FALSE,
        'none': boolean.FALSE,
    }

    length = len(expr)
    offset = 0
    while offset < length:
        tok = expr[offset]

        sym = tok.isalpha() or tok == '_'
        if sym:
            offset += 1
            while offset < length:
                char = expr[offset]
                if char.isalnum() or char in ('.', ':', '_'):
                    offset += 1
                    tok += char
                else:
                    break
            offset -= 1

        try:
            yield TOKENS[tok.lower()], tok, offset
        except KeyError:
            if sym:
                yield symbol_class(tok), tok, offset
            elif tok not in (' ', '\t', '\r', '\n'):
                raise TypeError("Unknown token %s at position %s." % (tok, offset))

        offset += 1
```

This prevents the use of `tokenize` and `StringIO`, and to me is easier to understand, since it's so simple. Users can always use their own tokenizers anyway.

**Comment by pombredanne** [comment]
@Kronuz actually CPython does not use `tokenize` for Pythoon compilation, `tokenize` is a Python version of the C tokenizer. That said, I am quite fine with your simpler version. It does not support comments, but who cares?
:+1:

**Comment by pombredanne** [comment]
@Kronuz Are you submitting a PR?

**Comment by pombredanne** [comment]
Fixed with #39 and 5a04e0c4ed105547510aed8cd26f3f4cbc7a2d10
